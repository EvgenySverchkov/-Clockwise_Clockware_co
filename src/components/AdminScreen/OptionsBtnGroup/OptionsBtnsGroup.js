import React from 'react';
import PropTypes from 'prop-types';

import DeleteBtn from "./DeleteBtn";
import LinkToEditForm from "./LinkToEditForm";
import ShowFullInfo from "./ShowFullInfo";

function OptionsBtnsGroup({deleteMasterById, itemObj}){
  return (
    <td className="text-center">
      <div className="dropdown">
          <button className="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
            Options
          </button>
          <div className="dropdown-menu" aria-labelledby="dropdownMenuLink">
            <DeleteBtn deleteMasterById={deleteMasterById} id={itemObj.id}/>
            <LinkToEditForm id={itemObj.id}/>
            <ShowFullInfo itemObj = {itemObj}/>
          </div>
      </div>
    </td>
  );
}

OptionsBtnsGroup.propTypes = {
  deleteMasterById: PropTypes.func,
  itemId: PropTypes.number
}

export default OptionsBtnsGroup;
