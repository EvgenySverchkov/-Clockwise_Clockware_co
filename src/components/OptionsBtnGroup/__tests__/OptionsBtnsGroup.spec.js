import React from "react";
import {create} from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configMockStore from 'redux-mock-store';

import OptionsBtnsGroup from "../OptionsBtnsGroup";

const mockStore = configMockStore();

it("Test of <OptionsBtnsGroup />", ()=>{
    const store = mockStore({})
    const mockProps = {
        deleteMasterById: jest.fn(), 
        itemObj: {
            someField: "Some Value"
        }
    }
    const component = create(
        <MemoryRouter initialEntries={["/"]}>
            <Provider store = {store}>
                <OptionsBtnsGroup {...mockProps}/>
            </Provider> 
        </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
});