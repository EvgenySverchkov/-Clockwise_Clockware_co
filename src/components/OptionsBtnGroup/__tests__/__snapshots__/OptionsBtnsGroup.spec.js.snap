// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <OptionsBtnsGroup /> 1`] = `
<td
  className="text-center"
>
  <div
    className="dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-secondary dropdown-toggle"
      data-toggle="dropdown"
      id="dropdownMenuButton"
      type="button"
    >
      Options
    </button>
    <div
      aria-labelledby="dropdownMenuLink"
      className="dropdown-menu"
    >
      <button
        className="dropdown-item deleteItemBtn"
        onClick={[Function]}
      >
        Delete
      </button>
      <a
        className="dropdown-item"
        href="/undefined"
        onClick={[Function]}
      >
        Edit
      </a>
      <button
        className="dropdown-item"
        data-target="#fullInfoModal"
        data-toggle="modal"
        onClick={[Function]}
        type="button"
      >
        Show full info
      </button>
    </div>
  </div>
</td>
`;
